### PR TITLE
Update Docs

### PR DESCRIPTION
**Pull request by chadell**

### CODE REVIEW

File: docs/user/design_development.md [hunk 1]
```diff
@@ -106,6 +106,10 @@ Design file specifies the Jinja template that should be used to produce the inpu
 
 Design files specifies a list of Jinja template that should be used to produce the input for the design builder. The builder will resolve the files' locations relative to the location of the design job class. Exactly one of `design_file` or `design_files` must be present in the design's Metadata. If `design_files` is used for a list of design templates, each one is evaluated in order. The same context and builder are used for all files. Since a single builder instance is used, references can be created in one design file and then accessed in a later design file.
 
+### `design_mode`
+
+It defines how this design is going to be tracked in Nautobot. By default, it uses the `CLASSIC` mode that has no tracking of objects created (for backwards compatibility), but it can use the new `DEPLOYMENT` mode that allows every design deployment to implement a full lifecycle.
```
Comment (abates) [inline docs/user/design_development.md:111]:
```suggestion
The `design_mode` indicates how this design's state will be tracked in Nautobot. By default, the design mode is set to CLASSIC` mode, which means that objects are created in an ad-hoc fashion and no change sets are created. Classic mode is how all designs worked prior to the introduction of the design lifecycle features. If a design is intended to be tracked as a deployment, then design mode should be set to `DEPLOYMENT` in order to implement the full lifecycle.
```

**Comment by abates** [comment]
I changed the base to `feature_delices_updates` just so I could see the changes from that branch. I added a few comments, but otherwise looks good to me.

File: docs/user/design_lifecycle.md [hunk 2]
```diff
@@ -1,67 +1,79 @@
...
-<!-- TODO: Add the screenshoot of the table of DesignInstance -->
+> Deletion is only possible if has been already decommissioned.
 
 ### Design Deployment Update
 
-This feature provides a means to re-run a design instance with different input data. Re-running the job will update the implemented design with the new changes: additions and removals.
+This feature provides a means to re-run a design deployment with different **input data**. Re-running the job will update the implemented design with the new changes: additions and removals.
 
-It leverages a complete tracking of previous design implementations and a function to combine the new design and previous design, to understand the changes to be implemented and the objects to be decommissioned (leveraging the previous decommissioning feature for only a specific object).
+It leverages a complete tracking of previous design implementations and a function to combine the new design and previous design, to understand the changes to be implemented and the objects to be decommissioned (more in next section).
```
[the remark this answers is not in the document]
Comment (chadell) [reply docs/user/design_lifecycle.md:54]: this is outdated after your refactor

File: docs/user/design_development.md [hunk 3]
```diff
@@ -106,6 +106,10 @@ Design file specifies the Jinja template that should be used to produce the inpu
 
 Design files specifies a list of Jinja template that should be used to produce the input for the design builder. The builder will resolve the files' locations relative to the location of the design job class. Exactly one of `design_file` or `design_files` must be present in the design's Metadata. If `design_files` is used for a list of design templates, each one is evaluated in order. The same context and builder are used for all files. Since a single builder instance is used, references can be created in one design file and then accessed in a later design file.
 
+### `design_mode`
+
+The `design_mode` indicates how this design's state will be tracked in Nautobot. By default, the design mode is set to CLASSIC` mode, which means that objects are created in an ad-hoc fashion and no change sets are created. Classic mode is how all designs worked prior to the introduction of the design lifecycle features. If a design is intended to be tracked as a deployment, then design mode should be set to `DEPLOYMENT` in order to implement the full lifecycle.
```
Comment (abates) [inline docs/user/design_development.md:111]:
```suggestion
The `design_mode` indicates how this design's state will be tracked in Nautobot. By default, the design mode is set to `CLASSIC` mode, which means that objects are created in an ad-hoc fashion and no change sets are created. Classic mode is how all designs worked prior to the introduction of the design lifecycle features. If a design is intended to be tracked as a deployment, then design mode should be set to `DEPLOYMENT` in order to implement the full lifecycle.
```

**Pull request merged**